akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -39,6 +46,15 @@ private[http2] object RequestParsing {
 
         headers += Http2StreamIdHeader(subStream.streamId)
 
+        if (serverSettings.remoteAddressHeader) {
```

Comment:
If you change the signature of the method to

```scala
def parseRequest(httpHeaderParser: HttpHeaderParser): Http2SubStream => HttpRequest
```

this calculation can be done only once per connection and then be reused for every request afterwards like this:

```scala
val remoteAddressHeader = 
   if (serverSettings.remoteAddressHeader) 
     OptionVal.Some { ... }
   else OptionVal.None

subStream => 

if (remoteAddressHeader.isDefined) headers += remoteAddressHeader.get
```